platform/android/export/export.cpp

```diff
@@ -768,6 +768,81 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 		}
 	}
 
+	void _write_tmp_manifest(const Ref<EditorExportPreset> &p_preset, bool p_give_internet, bool p_debug) {
+		String manifest_text =
+				"<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"
+				"<manifest xmlns:android=\"http://schemas.android.com/apk/res/android\"\n"
+				"    xmlns:tools=\"http://schemas.android.com/tools\">\n";
+
+		manifest_text += _get_screen_sizes_tag(p_preset);
+		manifest_text += _get_gles_tag();
+
+		Vector<String> perms;
+		_get_permissions(p_preset, p_give_internet, perms);
+		for (int i = 0; i < perms.size(); i++) {
+			manifest_text += vformat("    <uses-permission android:name=\"%s\" />\n", perms.get(i));
+		}
+
+		bool uses_xr = (int)(p_preset->get("xr_features/xr_mode")) == 1;
+		if (uses_xr) {
+			int dof_index = p_preset->get("xr_features/degrees_of_freedom"); // 0: none, 1: 3dof and 6dof, 2: 6dof
+			if (dof_index == 1) {
+				manifest_text += "    <uses-feature tools:node=\"replace\" android:name=\"android.hardware.vr.headtracking\" android:required=\"false\" android:version=\"1\" />\n";
+			} else if (dof_index == 2) {
+				manifest_text += "    <uses-feature tools:node=\"replace\" android:name=\"android.hardware.vr.headtracking\" android:required=\"true\" android:version=\"1\" />\n";
+			}
+			int hand_tracking_index = p_preset->get("xr_features/hand_tracking"); // 0: none, 1: optional, 2: required
+			if (hand_tracking_index == 1) {
+				manifest_text += "    <uses-feature tools:node=\"replace\" android:name=\"oculus.software.handtracking\" android:required=\"false\" />\n";
+			} else if (hand_tracking_index == 2) {
+				manifest_text += "    <uses-feature tools:node=\"replace\" android:name=\"oculus.software.handtracking\" android:required=\"true\" />\n";
+			}
+		}
+
+		String package_name = p_preset->get("package/unique_name");
+		manifest_text += vformat(
+				"    <instrumentation\n"
+				"        tools:node=\"replace\"\n"
+				"        android:name=\".GodotInstrumentation\"\n"
+				"        android:icon=\"@mipmap/icon\"\n"
+				"        android:label=\"@string/godot_project_name_string\"\n"
+				"        android:targetPackage=\"%s\" />\n",
+				package_name);
+
+		manifest_text +=
+				"    <application android:label=\"@string/godot_project_name_string\"\n"
+				"        android:allowBackup=\"false\" tools:ignore=\"GoogleAppIndexingWarning\"\n"
+				"        android:icon=\"@mipmap/icon\">)\n";
+
+		String plugins_names = get_plugins_names(get_enabled_plugins(p_preset));
+		if (!plugins_names.empty()) {
+			manifest_text += vformat("    <meta-data tools:node=\"replace\" android:name=\"plugins\" android:value=\"%s\" />\n", plugins_names);
+		}
+
+		if (uses_xr) {
+			manifest_text += "        <meta-data tools:node=\"replace\" android:name=\"com.samsung.android.vr.application.mode\" android:value=\"vr_only\" />\n";
+		}
+
```

Comment:
Should I add a line to remove this element if `uses_xr == false`?

Reply:
Yes